api/src/main/java/com/spotify/metrics/core/RemoteSemanticMetricRegistry.java

```diff
@@ -104,4 +104,9 @@
      * @return a new {@link RemoteMeter}
      */
     RemoteMeter meter(final MetricId name);
+
+    /**
+     * Wait for all metrics to be sent to the remote.
+     */
+    void waitForAllCalls();
```

Comment:
Should probably be named `shutdown`.

Should also take a timeout, or return a Future to permit callers to determine how long it is appropriate to wait. As it stands it could wait indefinitely which is not acceptable for a shutdown procedure.